tests/h/services/search_index/_queue_test.py

```diff
@@ -314,13 +322,51 @@ def test_deleting_multiple_jobs_with_the_same_annotation_id(
         counts = queue.sync(len(jobs))
 
         assert counts == {
-            Queue.Result.COMPLETED_UP_TO_DATE: 2,
+            Queue.Result.COMPLETED_UP_TO_DATE.format(tag="test_tag"): 2,
+            Queue.Result.COMPLETED_TAG_TOTAL.format(tag="test_tag"): 2,
             Queue.Result.COMPLETED_TOTAL: 2,
         }
         for job in jobs:
             assert job not in db_session.query(Job)
         batch_indexer.index.assert_not_called()
 
+    def test_metrics(self, factories, index, now, queue):
+        def add_job(indexed=True, updated=False, deleted=False, **kwargs):
+            annotation = factories.Annotation()
+            factories.SyncAnnotationJob(annotation=annotation, **kwargs)
+
+            if indexed:
+                index(annotation)
+
+            if updated:
+                annotation.updated = now + ONE_WEEK
+
+            if deleted:
+                annotation.deleted = True
+
+        add_job()
+        add_job(indexed=False)
+        add_job(updated=True)
+        add_job(deleted=True)
+        add_job(tag="tag_2", force=True)
+
+        counts = queue.sync(5)
+
+        assert counts == {
+            Queue.Result.SYNCED_MISSING.format(tag="test_tag"): 1,
+            Queue.Result.SYNCED_DIFFERENT.format(tag="test_tag"): 1,
+            Queue.Result.SYNCED_TAG_TOTAL.format(tag="test_tag"): 2,
+            Queue.Result.SYNCED_FORCED.format(tag="tag_2"): 1,
+            Queue.Result.SYNCED_TAG_TOTAL.format(tag="tag_2"): 1,
+            Queue.Result.SYNCED_TOTAL: 3,
+            Queue.Result.COMPLETED_UP_TO_DATE.format(tag="test_tag"): 1,
+            Queue.Result.COMPLETED_DELETED.format(tag="test_tag"): 1,
+            Queue.Result.COMPLETED_TAG_TOTAL.format(tag="test_tag"): 2,
+            Queue.Result.COMPLETED_FORCED.format(tag="tag_2"): 1,
+            Queue.Result.COMPLETED_TAG_TOTAL.format(tag="tag_2"): 1,
+            Queue.Result.COMPLETED_TOTAL: 3,
+        }
```

Comment:
I wanted to add one more test that runs it with a bunch of different jobs at once, and test that the resulting complex metrics reporting was right. I think this is the only test that covers reporting metrics for two different tags at once, too

Reply:
I think a big mixed up test that exercises everything at once is a good way to go about things like this. Yes you could spend the rest of your life writing individual test cases, but honestly it just makes it awful to maintain, and it isn't actually any more informative.